src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.html

```diff
@@ -47,15 +47,9 @@
 
 <ng-template #servicesTpl
              let-value="value">
-  <span *ngFor="let service of value; last as isLast">
-    <a class="service-link"
-       [routerLink]="[service.cdLink]"
-       [queryParams]="cdParams"
-       *ngIf="service.canRead">{{ service.type }}.{{ service.id }}</a>
-    <span *ngIf="!service.canRead">
-      {{ service.type }}.{{ service.id }}
-    </span>
-    <ng-container *ngIf="!isLast">, </ng-container>
+  <span *ngFor="let instance of value; last as isLast">
+    <span class="badge badge-background-primary" >{{ instance }}</span>
+    <ng-container *ngIf="!isLast">&nbsp;</ng-container>
```

Comment:
While this shows super attention to detail (kudos for that @nizamial09 😊 ), probably we can live with a trailing whitespace at the end (no one is going to notice it and it simplifies the template). As an alternative for badges, given whitespacing makes most sense for paragraphs/text, we could rely on CSS `.badge { margin-right: X}`.

Reply:
without space it looked congested to me. That's why I though I'll add one space there but you are right its better to do it with the css than using an extra html template. Hmmm, I'll clean it up at some point for sure.